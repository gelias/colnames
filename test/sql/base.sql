\set ECHO 0
BEGIN;
\set QUIET 1
\i sql/colnames.sql

SELECT colnames( ROW(1, 3, 'foo') );
SELECT colnames( ROW(c.*)::pg_collation ) FROM pg_collation c LIMIT 1;
SELECT colnames( NULL::pg_collation );

CREATE TYPE foo AS (id int, name text);
SELECT colnames( row(1, 'foo')::foo );

CREATE TYPE bar AS ("The Id" int, "My Name" text);
SELECT colnames( row(1, 'bar')::bar );

ROLLBACK;
