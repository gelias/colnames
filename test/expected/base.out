\set ECHO none
  colnames  
------------
 {f1,f2,f3}
(1 row)

                colnames                
----------------------------------------
 {objoid,classoid,objsubid,description}
(1 row)

                colnames                
----------------------------------------
 {objoid,classoid,objsubid,description}
(1 row)

 colnames  
-----------
 {id,name}
(1 row)

       colnames       
----------------------
 {"The Id","My Name"}
(1 row)

 colnames 
----------
 {}
(1 row)

